dev/com.ibm.ws.classloading.bells/src/com/ibm/ws/classloading/bells/internal/Bell.java

```diff
@@ -516,10 +543,27 @@ private static Class<?> findClass(final String implClass, final Library library,
         return service;
     }
 
-    private static Object createService(final Class<?> serviceType, final String libID, final URL fileUrl) {
+    private static Object createService(final Class<?> serviceType, final String libID, final URL fileUrl, final Map<String, String> properties) {
         Object service = null;
         try {
-            service = serviceType.newInstance();
+            if (properties == null) {
+                service = serviceType.newInstance();
```

Comment:
What if the service implementation only has a constructor that takes a `Map<String,String>`?  Here we always force them to have a no arg constructor?  Should we fall back to calling the `ctor(Map)` with an empty map if there is no, no-arg constructor available?

Reply:
For now I'd prefer to leave the solution as-is.  According to documentation, and some implicit convention mentioned in our design call, a service implementation class is required to provide an public zero-arg constructor. It's possible a user will ask for the behavior, and we can reconsider at that time.